src/bundle/Resources/public/scss/_inputs.scss

```diff
@@ -36,8 +39,59 @@
         border-color: $ibexa-color-dark-200;
     }
 
+    &--text {
+        @include ibexa-font-sans-serif();
+
+        font-size: $ibexa-text-font-size-medium;
+        color: $ibexa-color-dark;
+        line-height: calculateRem(21px);
+        height: calculateRem(48px);
+        padding-right: calculateRem(36px);
+        border: calculateRem(1px) solid $ibexa-color-dark-200;
+        border-radius: $ibexa-border-radius;
+        position: relative;
+
+        &:focus,
+        &.form-control:focus {
+            color: $ibexa-color-dark;
+        }
+
+        &:hover:not(:disabled) {
+            border-color: $ibexa-color-primary;
+        }
+
+        &:active:not(:disabled) {
+            border-color: $ibexa-color-dark;
+        }
+
+        &:disabled,
+        &.form-control:disabled {
+            color: $ibexa-color-dark-300;
+            background-color: $ibexa-color-light-300;
+        }
+
+        &.form-control.is-invalid,
+        &.form-control:invalid {
+            color: $ibexa-color-danger;
+            background-image: none;
+            border-color: $ibexa-color-danger;
+            background-color: $ibexa-color-danger-100;
+        }
+
+        &.ibexa-input--small {
+            height: calculateRem(32px);
+            padding-top: 0;
+            padding-bottom: 0;
+        }
+    }
+
     &--checkbox,
     &--radio {
+        position: relative;
+        display: inline-block;
+        vertical-align: middle;
+        margin-top: 0;
+
```

Comment:
why new line? Also I would move it uder appearance, but it's just my preferred order of properties